executables/collector/src/main/kotlin/de/cyface/collector/handler/MeasurementHandler.kt

```diff
@@ -0,0 +1,573 @@
+
+            // Etc.
+            val deviceType = request.getHeader(FormAttributes.DEVICE_TYPE.value)
+            val osVersion = request.getHeader(FormAttributes.OS_VERSION.value)
+            val applicationVersion = request.getHeader(FormAttributes.APPLICATION_VERSION.value)
+            val length = request.getHeader(FormAttributes.LENGTH.value).toDouble()
+            val modality = request.getHeader(FormAttributes.MODALITY.value)
+            RequestMetaData(
+                deviceId, measurementId, osVersion, deviceType, applicationVersion,
+                length, locationCount, startLocation, endLocation, modality, formatVersion
+            )
+        } catch (e: IllegalArgumentException) {
+            throw InvalidMetaData("Data was not parsable!", e)
+        } catch (e: NullPointerException) {
+            throw InvalidMetaData("Data was not parsable!", e)
+        }
+    }
+
+    /**
+     * Extracts the content range information from the request header and checks it matches the body size information
+     * from the header.
+     *
+     * @param request the request to check the header for
+     * @param bodySize the number of bytes to be uploaded
+     * @return the extracted content range information
+     * @throws Unparsable if the content range does not match the body size information
+     */
+    @Throws(Unparsable::class)
+    private fun contentRange(request: HttpServerRequest, bodySize: Long): ContentRange {
+
+        // The client informs what data is attached: `bytes fromIndex-toIndex/totalBytes`
+        val contentRangeString = request.getHeader("Content-Range")
+        if (!contentRangeString.matches(Regex("bytes [0-9]+-[0-9]+/[0-9]+"))) {
+            throw Unparsable(String.format("Content-Range request not supported: %s", contentRangeString))
+        }
+        val startingWithFrom = contentRangeString.substring(6)
+        val dashPosition = startingWithFrom.indexOf('-')
+        Validate.isTrue(dashPosition != -1)
+        val from = startingWithFrom.substring(0, dashPosition)
+        val startingWithTo = startingWithFrom.substring(dashPosition + 1)
+        val slashPosition = startingWithTo.indexOf('/')
+        Validate.isTrue(slashPosition != -1)
+        val to = startingWithTo.substring(0, slashPosition)
+        val total = startingWithTo.substring(slashPosition + 1)
+        val contentRange = ContentRange(from, to, total)
+
+        // Make sure content-length matches the content range (to-from+1)
+        if (bodySize != contentRange.toIndex.toLong() - contentRange.fromIndex.toLong() + 1) {
+            throw Unparsable(
+                String.format(
+                    "Upload size (%d) does not match content rang of header (%s)!",
+                    bodySize, contentRange
+                )
+            )
+        }
+        return contentRange
+    }
+
+    /**
+     * Creates a new upload temp file and calls
+     * [.acceptUpload].
```

Comment:
strange line break

Reply:
This is how a link is formatted in Kotlin. In java this includes all the parameters and thus requires a new line. The converter just took this over.

Reply:
I see. Thanks for clarification, I have to get used to the Kotlin world I guess.